### PR TITLE
chore: improve checking if the pipelinerun succeeded in e2e tests

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
chore: improve checking if the pipelinerun succeeded in e2e tests

previously the script waited for the state of pipelinerun, even the pipelinerun has already succeeded. This commit updates to behaviour to end immediatelly if the pipelinerun is finished

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by jcanocan** [comment]
Nice addition!
/lgtm

File: automation/e2e-pipelines/example-pipelines-test.sh [hunk 1]
```diff
@@ -11,16 +11,11 @@ EOF
...
-  oc wait -n ${namespace} --for=condition=Succeeded=False pipelinerun -l pipelinerun="$1"-run --timeout=60m && exit 1 &
-  failure_pid=$!
-
-  if wait -n $success_pid $failure_pid ; then
+  oc wait -n ${namespace} --for=condition=Succeeded=True pipelinerun -l pipelinerun="$1"-run --timeout=60m
+  if ! $?; then
     echo "Pipelinerun $1 succeeded"
   else
-    echo "Pipelinerun $1 failed"
+    echo "Pipelinerun $1 timeouted"
```
Comment (0xFelix) [inline automation/e2e-pipelines/example-pipelines-test.sh:18]:
```suggestion
    echo "Pipelinerun $1 timed out"
```

File: automation/e2e-pipelines/example-pipelines-test.sh [hunk 2]
```diff
@@ -11,16 +11,11 @@ EOF
...
 function wait_for_pipelinerun() {
-  oc wait -n ${namespace} --for=condition=Succeeded=True pipelinerun -l pipelinerun="$1"-run --timeout=60m &
-  success_pid=$!
-
-  oc wait -n ${namespace} --for=condition=Succeeded=False pipelinerun -l pipelinerun="$1"-run --timeout=60m && exit 1 &
-  failure_pid=$!
-
-  if wait -n $success_pid $failure_pid ; then
+  oc wait -n ${namespace} --for=condition=Succeeded=True pipelinerun -l pipelinerun="$1"-run --timeout=60m
+  if ! $?; then
```
Comment (0xFelix) [inline automation/e2e-pipelines/example-pipelines-test.sh:15]: Not sure if it improves readability, but you could put the command directly into the if condition.

Comment (ksimon1) [reply automation/e2e-pipelines/example-pipelines-test.sh:15]: It will work for sure, but I would keep it as it is

**Review by 0xFelix — COMMENTED** [review]
If condition Succeeded is False that means it is still running or it failed? Ideally it would exit as soon as it knows the PipelineRun has failed, without waiting for the timeout.

File: automation/e2e-pipelines/example-pipelines-test.sh [hunk 3]
```diff
@@ -11,16 +11,26 @@ EOF
...
 
 function wait_for_pipelinerun() {
-  oc wait -n ${namespace} --for=condition=Succeeded=True pipelinerun -l pipelinerun="$1"-run --timeout=60m &
-  success_pid=$!
-
-  oc wait -n ${namespace} --for=condition=Succeeded=False pipelinerun -l pipelinerun="$1"-run --timeout=60m && exit 1 &
-  failure_pid=$!
+  sample=10
+  current_time=0
+  timeout=2000
```
Comment (0xFelix) [inline automation/e2e-pipelines/example-pipelines-test.sh:16]: Make those local vars to the func?

File: automation/e2e-pipelines/example-pipelines-test.sh [hunk 4]
```diff
@@ -11,16 +11,26 @@ EOF
...
+    fi
+    (( current_time+=sample ))
+    sleep $sample
+  done
 
-  if wait -n $success_pid $failure_pid ; then
-    echo "Pipelinerun $1 succeeded"
-  else
-    echo "Pipelinerun $1 failed"
+  if [ $current_time -eq $timeout ]; then
```
Comment (0xFelix) [inline automation/e2e-pipelines/example-pipelines-test.sh:32]: Use `-ge` for greate than equal?

Comment (0xFelix) [inline automation/e2e-pipelines/example-pipelines-test.sh:32]: I would make this block part of the loop.

**Review by 0xFelix — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**